### PR TITLE
Authenticator setup

### PR DESCRIPTION
**Pull request by AtomicZero**
added api.js to routes folder
updated auth and authenticator to server index.js
added middleware folder with authenticator.js for user data

### CODE REVIEW

**Comment by kepop1** [comment]
Be careful making a branch that implements the same code as the other branch - as PR#4 will now be merged in and could conflict with similar code that is also changed in this PR. Hopefully it's nice and smooth though 🤞

**Review by kepop1 — APPROVED** [review]

File: src/index.js [hunk 1]
```diff
@@ -11,8 +13,9 @@ app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
 app.use(cors());
-
-app.use('/api', routes);
+app.use('/auth', auth);
+app.use(authenticateUser);
+app.use('/api', api);
```
Comment (surajverma2587) [inline src/index.js:18]:
```suggestion
app.use('/api', authenticateUser, api);
```

This allows all calls to the `/api` route to be authenticated

File: src/index.js [hunk 2]
```diff
@@ -11,8 +13,9 @@ app.use(express.json());
 app.use(express.urlencoded({ extended: true }));
 
 app.use(cors());
-
-app.use('/api', routes);
+app.use('/auth', auth);
+app.use(authenticateUser);
```
Comment (surajverma2587) [inline src/index.js:17]:
```suggestion

```

Delete this

Comment (AtomicZero) [reply src/index.js]: updated lines

**Review by surajverma2587 — APPROVED** [review]

**Pull request merged**